R/RLearner_regr_randomForest.R

```diff
@@ -90,43 +112,53 @@ getOOBPredsLearner.regr.randomForest = function(.learner, .model) {
 }
 
 # Computes brute force or noisy bootstrap
-# Set ntree = ntree.for.se for the brute force bootstrap
-# Set ntree.for.se << ntree for the noisy bootstrap (mc bias corrected)
-bootstrapStandardError = function(.learner, .model, .newdata, ...) {
-  pred.all.boot = lapply(getLearnerModel(.model$learner.model), function(x)
-    predict(x$learner.model, newdata = .newdata, predict.all = TRUE)$individual)
-  B = .learner$par.vals$se.boot
-  R = .learner$par.vals$ntree.for.se
-  M = .learner$par.vals$ntree
-  bias = ((1 / R) - (1 / M)) / (B * R * (R - 1)) *
-    rowSums(matrix(sapply(pred.all.boot, function(p) rowSums((p - mean(p))^2)),
-                   nrow = nrow(.newdata), ncol = R, byrow = TRUE))
-  pred = getPredictionResponse(predict(.model$learner.model, newdata = .newdata))
-  pred.boot = lapply(getLearnerModel(.model$learner.model), predict, newdata = .newdata, ...)
-  pred.boot = extractSubList(pred.boot, c("data", "response"))
-  if (is.vector(pred.boot)) {
-    pred.boot = matrix(pred.boot, nrow = nrow(.newdata), ncol = R, byrow = TRUE)
-  }
-  var.boot = apply(pred.boot, 1, var) - bias
-  var.boot[var.boot <= 0] = 0
-  cbind(pred, sqrt(var.boot))
+# Set ntree = se.ntree for the brute force bootstrap
+# Set se.ntree << ntree for the noisy bootstrap (mc bias corrected)
+bootstrapStandardError = function(.learner, .model, .newdata,
+  se.ntree = 100L, se.boot = 50L, ...) {
+  single.model = getLearnerModel(.model)$single.model #get raw RF model
+  bagged.models = getLearnerModel(getLearnerModel(.model)$bagged.models) #get list of unbagged mlr models
+  pred.single = predict(single.model, newdata = .newdata, ...)
```

Comment:
these lines should all be moved to predictLearner. the se.funs should only compute and return the se as a single vec